### DISCUSSION TITLE
Any plans to revamp tailwindcss-typography in 4.0 or beyond?

### DISCUSSION THREAD

**Discussion by EmmanuelOga** [tailwindlabs/tailwindcss#15305] [Ideas]
First, thank you for all the incredible work on Tailwind, it’s been a game-changer for my CSS workflow!

With the upcoming 4.0 release, I was wondering if there are plans to revamp or update the [Tailwind Typography plugin](https://github.com/tailwindlabs/tailwindcss-typography)?

While the plugin adds great functionality, I’ve found it challenging compared to the rest of Tailwind.

Some friction points I’ve experienced:

* Customizing link colors outside the predefined color schemes took me hours. I eventually gave up in frustration and settled for one of the “close enough” presets. Here's an ugly fix that I've been using:

```tsx
 const overrideProse = { style: "color: var(--color-foreground);" };
 <a {...overrideProse} href="..." >My Link</a>
 ```

* The plugin heavily relies on CSS variables, which sometimes interfere with regular classes. For instance, I had to use `!important` to reset margins in a few places (e.g., `<h2 class="mt-0!">`).

* Modifying text sizes inside a `<div class="prose">` wrapper has also been unexpectedly tricky. In one case, I wanted to override the default font size, so naturally I tried `<span class="text-xs">` (didn’t work), then `<span class="text-xs!">` (still no luck), and finally `<span class="not-prose text-xs!">`, but still, it did not work. My final solution was: `<span style="font-size: 1rem">`, which worked as expected.

I may be missing something, but it feels like the plugin’s current behavior interferes with core functionality in ways that are hard to debug, and the _quality of the docs is nowhere near the core Tailwind Docs_.

These challenges make the plugin feel less aligned with Tailwind’s simplicity and ease of use. I’m curious if others have reported similar issues and whether improvements are planned to address these pain points.

Thanks again for all your hard work! I’m excited to see what’s next.

**Comment by KorigamiK** [comment]
I agree, the customization is just not the same as before. Overriding styles is definitely not the best. Having configuration actually change the styles is ideal.

**Comment by adamwathan** [comment]
Yeah we're going to revisit this at some point — I really hate the typography plugin, especially the customization story, so I want to come up with an approach that's simpler and easier to work with.

I honestly feel like the absolute best approach is just giving you some CSS you can copy and paste and edit yourself instead of trying to override things or configure things, but need to experiment and figure it out.

**Reply by vgarmes** [reply]
This is what I ended up doing. I copied the generated CSS from the plugin and pasted it into a separate CSS file. This way, I can configure it however I want (especially the colors, to align them with the rest of my site) and even remove any classes I don’t need

**Comment by khadershafeeq** [comment]
How to configure typography in tailwindcss v4. I am using nextjs

**Reply by wongjn** [reply]
See the readme: https://github.com/tailwindlabs/tailwindcss-typography?tab=readme-ov-file#adding-custom-color-themes

**Comment by danmilward** [comment]
Hello. I'd just like to bump this. We too use Typography Plugin on StoryMate and the overhaul being described by @adamwathan sounds like just what we need :)

**Comment by ynamite** [comment]
bump
